python/langsmith/testing/_internal.py

```diff
@@ -684,23 +653,24 @@ def end_run(
             run_tree=run_tree,
             example_id=example_id,
             outputs=outputs,
-            end_time=end_time,
+            reference_outputs=reference_outputs,
             pytest_plugin=pytest_plugin,
             pytest_nodeid=pytest_nodeid,
         )
 
     def _end_run(
-        self, run_tree, example_id, outputs, end_time, pytest_plugin, pytest_nodeid
+        self,
+        run_tree,
+        example_id,
+        outputs,
+        reference_outputs,
+        pytest_plugin,
+        pytest_nodeid,
     ) -> None:
-        # TODO: remove this hack so that run durations are correct
         # Ensure example is fully updated
-        self.wait_example_updates(example_id)
-        # Ensure that run end time is after example modified at.
-        example_modified_at = self.get_version(example_id=example_id) or end_time
-        end_time = max(example_modified_at, end_time)
```

Comment:
i think we still want to keep this part. just because sync_example returns doesn't mean the example has actually been updated in the backend, stuff is queued and can take a bit

Reply:
Ah ok interesting

Reply:
when i didn't have this i was pretty frequently running into issue that example modified time was after run end time. maybe something else that you've changed will alleviate that, but worth really testing

Reply:
note to test it make sure you're running on a new dataset and/or that your tests are updating the examples in some way. otherwise you won't be able to tell

Reply:
Seems to be working ok for me